File: 24_김수빈/10872.c
// 작성일: 2021년 10월 31일
// 작성자: 김수빈
// 프로그램명: 팩토리얼
#include <stdio.h>

int factorial(int num) {
	if (num == 0)
		return 1;
	else
		return num * factorial(num - 1);
}

int main() {
	int num;

	scanf_s("%d", &num);

	printf("factorial(%d) = %d\n", num, factorial(num));
	return 0;
}

File: 12_이예빈/10872.py
# 0보다 크거나 같은 정수 N이 주어진다. 이때, N!을 출력하는 프로그램을 작성하시오.

# factorial 재귀함수
def factorial(n):
    if n==1:
        return 1
    else:
        return n*factorial(n-1)
    
n = int(input())
print(factorial(n))